src/librustdoc/html/static/search.js

```diff
@@ -988,19 +988,43 @@ window.initSearch = function(rawSearchIndex) {
                 }
                 length += 1;
 
-                output += "<a class=\"result-" + type + "\" href=\"" + item.href + "\">" +
-                          "<div><div class=\"result-name\">" +
-                          (item.is_alias === true ?
-                           ("<span class=\"alias\"><b>" + item.alias + " </b></span><span " +
-                              "class=\"grey\"><i>&nbsp;- see&nbsp;</i></span>") : "") +
-                          item.displayPath + "<span class=\"" + type + "\">" +
-                          name + "</span></div><div>" +
-                          "<span class=\"desc\">" + item.desc +
-                          "&nbsp;</span></div></div></a>";
+                var link = document.createElement("a");
+                link.className = "result-" + type;
+                link.href = item.href;
+
+                var wrapper = document.createElement("div");
+                var resultName = document.createElement("div");
+                resultName.className = "result-name";
+
+                if (item.is_alias) {
+                    var alias = document.createElement("span");
+                    alias.className = "alias";
+
+                    var bold = document.createElement("b");
+                    bold.innerText = item.alias;
+                    alias.appendChild(bold);
+
+                    alias.innerHTML += "<span class=\"grey\"><i>&nbsp;- see&nbsp;</i></span>";
+
+                    resultName.appendChild(alias);
+                }
+                resultName.innerHTML += item.displayPath + "<span class=\"" + type + "\">" +
+                    name + "</span>";
+                wrapper.appendChild(resultName);
+
+                var description = document.createElement("div");
```

Comment:
```suggestion
                var description = document.createElement("div");
                description.className = "result-description";
```
Adding this would simplify the CSS in the mobile variant

Reply:
Let's do it in another PR, this one is big enough, I prefer style changes to be put on their own.